odoo/addons/base/ir/ir_actions_report.py

```diff
@@ -180,73 +146,50 @@ def unlink_action(self):
     # Main report methods
     #--------------------------------------------------------------------------
     @api.multi
-    def retrieve_attachment(self, record_id, attachment_name=None):
+    def retrieve_attachment(self, record_id):
```

Comment:
The name `record_id` is confusing: it suggests that it should be a record ID. I recommend to use `record` instead.